armory/validation/test_config/test_model.py

```diff
@@ -0,0 +1,135 @@
+                test_input_shape[np.equal(test_input_shape, None)] = 32
+                warnings.warn(
+                    f"Model shape given as {input_shape}. Assuming {test_input_shape}"
+                )
+                input_shape = test_input_shape
+        elif isinstance(model, ObjectDetectorMixin):
+            input_shape = np.array((32, 32, 3))
+            warnings.warn(
+                "Object detector model does not specify input shape. Assuming (32, 32, 3)"
+            )
+        elif isinstance(model, SpeechRecognizerMixin):
+            warnings.warn(
+                "Speech recognition model does not specify input shape. Assuming (10000,)"
+            )
+            input_shape = np.array((10000,))
+        return input_shape
+
+    def _get_clip_values(self, model):
+        if isinstance(model, ClassifierMixin) or isinstance(model, ObjectDetectorMixin):
+            assert (
+                model.clip_values is not None
+            ), "Clip values not provided. Clip values are required to keep input values within valid range"
+            clip_min, clip_max = model.clip_values
+        elif isinstance(model, SpeechRecognizerMixin):
+            clip_min, clip_max = (-1.0, 1.0)
+        return clip_min, clip_max
+
+    def _get_test_input(self, input_shape, clip_min, clip_max):
+        test_sample = np.random.randn(1, *input_shape).astype(np.float32)
+        test_sample = np.clip(test_sample, clip_min, clip_max)
+        return test_sample
+
+    def _get_test_ground_truth(self, model, test_output):
+        if isinstance(model, ClassifierMixin) or isinstance(model, ObjectDetectorMixin):
+            test_ground_truth = test_output
+            if (
+                isinstance(test_output, np.ndarray)
+                and test_output.ndim == 2
+                and test_output.shape[1] > 1
+            ):
+                test_ground_truth = np.zeros((1, model.nb_classes))
+                test_ground_truth[:, 0] = 1
+        elif isinstance(model, SpeechRecognizerMixin):
+            test_ground_truth = np.array(["HELLO"])
+        return test_ground_truth
+
+    def test_model(self, model_config):
+
+        # verify model loads without error
+        model_config = json.loads(model_config)
+        model = self._get_model(model_config)
+        input_shape = self._get_input_shape(model)
+
+        # verify model clip values broadcast to input shape
+        clip_min, clip_max = self._get_clip_values(model)
+        np.broadcast_to(clip_min, input_shape)
+        np.broadcast_to(clip_max, input_shape)
+
+        # verify model forward pass
+        test_input = self._get_test_input(input_shape, clip_min, clip_max)
+        copy_input = test_input.copy()
+        test_output = model.predict(test_input)
+        assert np.all(
+            test_input == copy_input
+        ), "Model prediction overwrites the input x value"
+        if hasattr(model, "nb_classes"):
```

Comment:
This is minor, but `test_output.shape` will fail when `test_output` is not a numpy array, which is the case for object detection. I suppose the ART object detectors don't have a `nb_classes` attribute, so this conditional block likely won't be reached, but there would be nothing stopping a user from adding a `nb_classes` attribute in their custom detector